### ISSUE TITLE
config.login not working with dev version

### ISSUE THREAD

**Issue by havok2063** [sdss/marvin#565]
**Describe the bug**
After pip upgrading to dev marvin, the `config.login` returns "you must have collaboration access to login"

**Expected behavior**
login should return a valid authentication token

**Comment by albireox** [comment]
Could this be solved by changing

```python
assert config.access == 'collab', 'You must have collaboration access to login.'
```

with

```python
 config.access = 'collab'
```
?

That seems to work fine for me.

**Comment by havok2063** [comment]
It's unclear to me what the original problem here was.  This was from the vetting and no original code, or error traceback was provided.  This was reported by Kate and that's the only description I have for it.  @bretthandrews might remember more if this occurred during the crash day.

**Comment by bretthandrews** [comment]
Kate didn't use the conda environment yml file but instead upgraded her marvin version via pip. I haven't tried to reproduce it yet.

**Comment by albireox** [comment]
I think the problem is that if you're in public mode you have to manually change to `collab` and then login. It seems that login should change the access mode but I may be missing some subtlety here.

**Comment by havok2063** [comment]
That's the intended behaviour.  For public users, they should 1.) not be able to switch to `collab` access and 2.) not be able to use the `login` method.  The default mode of marvin is `public` unless someone has a validated `netrc` file and sets their `check_access: True` in their custom config.  Then it will automatically switch to `collab` access.  Otherwise, the user must switch over manually.  

In public mode, there is no need to login at all.  That should also be true when in `collab` access mode and you try to access a DR release.  But I can't remember if I coded for that last point, so it may not be the case.

**Comment by havok2063** [comment]
I think this is now fixed via #575.  This can be closed after attempted reproduction.

**Comment by bretthandrews** [comment]
Works now on non-dev version
